File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> row(rowIndex + 1);
        row[0] = 1;
        row[rowIndex] = 1;

        for (int i = 1; i < rowIndex; ++i) {
            row[i] = binCo(rowIndex, i);
        }

        return row;
    }
private:
    long long binCo(int n, int k) {
        long long result = 1;

        for (int i = 1; i <= k; ++i) {
            result = result * (n - k + i) / i;
        }

        return result;
    }
};

File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.py
class Solution:
    def getRow(self, rowIndex: int) -> list[int]:
        row = [1] * (rowIndex + 1)

        for i in range(1, rowIndex):
            row[i] = self.binCo(rowIndex, i)
            row[rowIndex - i] = row[i]
        
        return row
    
    def binCo(self, n: int, k: int) -> int:
        total = 1
        for i in range(1, k + 1):
            total = total * (n - k + i) // i
        
        return total
        